src/textinput/TextInput.ts

```diff
@@ -0,0 +1,37 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { FormLabelMixin, FormLabelMixinProperties } from '@dojo/widget-core/mixins/FormLabel';
+import { v } from '@dojo/widget-core/d';
+import * as css from './styles/textinput.css';
+
+/**
+ * @type TextInputProperties
+ *
+ * Properties that can be set on a TextInput component
+ *
+ * @property type			Input type, e.g. text, email, tel, etc.
+ * @property onInput	Called when the node's 'input' event is fired
+ */
+export interface TextInputProperties extends ThemeableProperties, FormLabelMixinProperties {
+	type?: string;
+	onInput?(event: Event): void;
+}
+
+@theme(css)
+export default class TextInput extends ThemeableMixin(FormLabelMixin(WidgetBase))<TextInputProperties> {
+	onInput(event: Event) {
+		this.properties.onInput && this.properties.onInput(event);
+	}
+
+	render() {
```

Comment:
@tomdye: The discussion I had with @matt-gadd was based on the assumption that there was no easy way to invert control to the parent for form controls and that we'd have to break our unidirectional flow approach, but that's not the case. Using `initialValue` vs. `value` would work the same way, but the former wouldn't allow our form controls to invert control to the parent like all other components. We should be using `value` + `onInput` + updating parent state like all other components currently do.